src/lib/pickles/impls.ml

```diff
@@ -36,11 +40,15 @@ module Step = struct
       Field.t * Boolean.var
 
     let forbidden_shifted_values =
-      forbidden_shifted_values ~size_in_bits:Constant.size_in_bits
-        ~modulus:(Wrap_impl.Bigint.to_bignum_bigint Constant.size) ~f:(fun x ->
-          let hi = test_bit x (Field.size_in_bits - 1) in
-          let lo = B.shift_right x 1 in
-          (Impl.Bigint.(to_field (of_bignum_bigint lo)), hi))
+      let f x =
```

Comment:
Fixing this changes the circuit. Please make sure that this preserves the existing behaviour, so that we can still connect to the network.

Reply:
What do you mean by "make sure" here? This was passing CI, but is there anything else I can test?